consensus/parlia/parlia.go

```diff
@@ -1979,7 +1985,36 @@ func (p *Parlia) applyTransaction(
 		*receivedTxs = (*receivedTxs)[1:]
 	}
 	state.SetTxContext(expectedTx.Hash(), len(*txs))
-	gasUsed, err := applyMessage(msg, state, header, p.chainConfig, chainContext)
+
+	// Create a new context to be used in the EVM environment
+	context := core.NewEVMBlockContext(header, chainContext, nil)
+	// Create a new environment which holds all relevant information
+	// about the transaction and calling mechanisms.
+	vmenv := vm.NewEVM(context, vm.TxContext{Origin: msg.From(), GasPrice: big.NewInt(0)}, state, p.chainConfig, vmConfig)
```

Comment:
this one change the code logic, the `vmConfig` was empty before, but now it will use the `vmConfig` passed from uplayer.
``` previous
vmenv := vm.NewEVM(context, vm.TxContext{Origin: msg.From(), GasPrice: big.NewInt(0)}, state, chainConfig, vm.Config{})
```
Should be careful.

Reply:
That's true.

I can pass the `tracer` around instead of a pre-built `vm.Config` instance and I could change back to `vm.Config{Tracer: tracer}`. Would you prefer that?

Reply:
@zzzckck